### PR TITLE
3159 jest starter branch

### PR DESCRIPTION
**Pull request by Dananji**
This PR fixes avalonmediasystem/avalon#3159

There are more functions, which are dependent on other functions which can't be mocked with Jest at the moment. With a little bit of refactoring in the future, we will be able to write tests for them as well.

### CODE REVIEW

File: src/services/StructuralMetadataUtils.test.js [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+    expect(value).toHaveProperty('begin', '00:00:01');
+    expect(value).toHaveProperty('end', '00:00:02');
+    expect(value).toHaveProperty('label', 'Tester');
+  });
+
+  describe('tests new time overlaps existing time ranges', () => {
+    const allSpans = smu.getItemsOfType('span', testData);
+    test('time == 00:00:00.00 (before the first timespan)', () => {
+      const time = '00:00:00.00';
+      expect(smu.doesTimeOverlap(time, allSpans)).toBeTruthy();
```
Comment (adamjarling) [inline src/services/StructuralMetadataUtils.test.js]: This is my fault, but the function title `doesTimeOverlap` seems backwards from what the function returns:)  I might change that.

File: src/services/StructuralMetadataUtils.test.js [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+      ]);
+    });
+  });
+
+  describe('validates order of begin and end times', () => {
+    test('begin time < end time', () => {
+      const begin = '00:00:10.30';
+      const end = '00:10:00.30';
+      const vlaue = smu.validateBeforeEndTimeOrder(begin, end);
+      expect(vlaue).toBeTruthy();
```
Comment (adamjarling) [inline src/services/StructuralMetadataUtils.test.js]: Small typo in these tests for `vlaue`... if you want to fix great and if not no big deal.

File: src/services/StructuralMetadataUtils.test.js [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+    test('begin time > end time', () => {
+      const begin = '00:10:00.30';
+      const end = '00:00:10.30';
+      const vlaue = smu.validateBeforeEndTimeOrder(begin, end);
+      expect(vlaue).toBeFalsy();
+    });
+  });
+
+  describe('tests deleting list item', () => {
+    test('deletes a timespan from metadata structure', () => {
```
Comment (adamjarling) [inline src/services/StructuralMetadataUtils.test.js]: Another common pattern you'll see in JS unit testing is using a `beforeEach()` or `beforeAll()` function in a group of tests, which in this case could grab a fresh reference to `testData` before running each test.  It's a way to not mutate an object when running multiple assertions against some mock object.  Or to re setup your data before running multiple tests against it.

File: src/test/TestStructure.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      },
+      {
+        type: 'div',
+        label: 'Scene 2',
+        id: '123a-456b-789c-3d',
+        items: []
+      }
+    ]
+  }
+];
```
Comment (adamjarling) [inline src/test/TestStructure.js]: Nice job organizing this into it's own `/tests` folder.  Keeps it clean when I first look at it.

File: src/services/StructuralMetadataUtils.test.js (hunk 1 above)
Comment (Dananji) [reply src/services/StructuralMetadataUtils.test.js]: @adamjarling Oh yes. When I first wrote this I asserted the values the other way :)

File: src/services/StructuralMetadataUtils.test.js (hunk 3 above)
Comment (Dananji) [reply src/services/StructuralMetadataUtils.test.js]: @adamjarling Oh yeah, I'll try that. Thanks :+1:

**Comment by adamjarling** [comment]
@Dananji Even though the latest commit message is a little weird, what it does is install a very popular, common, helper library called Enzyme (https://airbnb.io/enzyme/), which gets us past the current `yarn test` error of the `<App />` component.  

We'll need Enzyme when we start testing individual components.  It basically allows you to test a component in isolation (through it's `shallow()` function), and not render all it's child components.  The goal with this approach is writing tests for only what individual components do.   At some point we'll write some integration tests as well to round it out.

File: src/services/StructuralMetadataUtils.test.js (hunk 3 above)
Comment (Dananji) [reply src/services/StructuralMetadataUtils.test.js]: @adamjarling I used `beforeEach()` to initialize the `testData` array, but still gets the same issue of changes in the original array when testing `deleteListItem()`. 
I think I should keep this for now, and do something else.

**Comment by Dananji** [comment]
@adamjarling The latest commit (3716f65) was my attempt to initialize test data with the use of `beforeEach()`, which did not work with deleting list item function (still changes the original test data structure). I committed this, so that you can look at it and give any feedback.

**Pull request merged**